mixed-content/generic/mixed-content-test-case.js

```diff
@@ -0,0 +1,224 @@
+function MixedContentTestCase(scenario, description, sanityChecker) {
+  var insecureProtocol = "http:"
+  var secureProtocol = "https:"
+
+  var sameOriginHost = location.hostname;
+  var crossOriginHost = "{{domains[www1]}}";
+
+  var insecurePort = getNormalizedPort(parseInt("{{ports[http][0]}}"));
+  var securePort = getNormalizedPort(parseInt("{{ports[https][0]}}"));
+
+  var resourcePath = "/mixed-content/generic/expect.py";
+
+  var endpoint = {
+    "same-origin":
+      location.origin + resourcePath,
+    "same-host-https":
+      secureProtocol + "//" + sameOriginHost + securePort + resourcePath,
+    "same-host-http":
+      insecureProtocol + "//" + sameOriginHost + insecurePort + resourcePath,
+    "cross-origin-https":
+      secureProtocol + "//" + crossOriginHost + securePort + resourcePath,
+    "cross-origin-http":
+      insecureProtocol + "//" + crossOriginHost + insecurePort + resourcePath
+  };
+
+  var resourceInvoker = {
```

Comment:
Renamed to resourceMap since the invoking method is now called `requestResource`:

```
  var requestResource = resourceMap[scenario.subresource];
```